generator/src/main/java/com/scottlogic/deg/generator/violations/filters/ConstraintTypeFilter.java

```diff
@@ -0,0 +1,38 @@
+package com.scottlogic.deg.generator.violations.filters;
+
+import com.scottlogic.deg.generator.constraints.Constraint;
+import com.scottlogic.deg.generator.constraints.grammatical.AndConstraint;
+import com.scottlogic.deg.generator.constraints.grammatical.ConditionalConstraint;
+
+public class ConstraintTypeFilter implements ViolationFilter{
+    private final Class constraintType;
+    public ConstraintTypeFilter(Class constraintType){
+        this.constraintType = constraintType;
+    }
+
+    @Override
+    public boolean accept(Constraint constraint) {
+        if (isConstraintType(constraint)) {
+            return false;
+        }
+        if (constraint instanceof ConditionalConstraint){
+            ConditionalConstraint conditional = (ConditionalConstraint) constraint;
+            if (isConstraintType(conditional.whenConditionIsTrue)){
+                return false;
+            }
+            if (conditional.whenConditionIsTrue instanceof AndConstraint){
```

Comment:
GrammaticalConstraint doesn't currently have a getSubConstraints method.

maybe it should do.

But if it did, i'm not entirely sure getting subconstraints and rejecting would be correct for or constraints?
but maybe it would be good enough for now